lib/Backends/NNPI/NNPIResource.cpp

```diff
@@ -192,8 +307,36 @@ bool NNPIResource::init(const NNPIObjectName name,
                                           deviceResource_, &copyCommand_),
         "Failed to create NNPI copy command (output)");
     break;
+  case ResourceUsage::P2POutput:
+    LOG_AND_RETURN_IF_NOT(ERROR, users, "Missing resource users", false);
+    for (auto reader : users->readers) {
+      if (reader && reader->getDevice() != device) {
+        p2pDevice_ = reader->getDevice();
+        p2pDeviceResource_ = reader->getDeviceResource();
+      }
+    }
+    LOG_AND_RETURN_IF_NOT(ERROR,
+                          (p2pDevice_ != NNPI_INVALID_NNPIHANDLE) &&
+                              (p2pDeviceResource_ != NNPI_INVALID_NNPIHANDLE),
+                          "Can't find p2p counterpart", false);
+    LOG_NNPI_INF_IF_ERROR_RETURN_FALSE(
+        nnpiCopyCommandCreateDeviceToDevice(p2pDevice_, p2pDeviceResource_,
+                                            device_, deviceResource_,
+                                            &copyCommand_),
+        "Failed to create NNPI copy command (p2p output)");
+    break;
+  case ResourceUsage::P2PInput:
+    // Fallthrough - the device resource is copied by the writer in the
+    // preceding context.
   case ResourceUsage::StaticInputResource:
-    // Fallthrough.
+    // Fallthrough - the device resource doesn't need to be updated before
+    // inference.
+  case ResourceUsage::DRTInput:
+    // Fallthrough - the device resource doesn't need to be updated before
+    // inference.
+  case ResourceUsage::DRTOutput:
+    // Fallthrough - the device resource doesn't need to be copied out after
+    // inference.
```

Comment:
nit: can we add explicit breaks to these? Feels like it could be error prone if someone adds something to a case here in the future but doesn't realize others will fall in (even if it's something simple like logging, as it doesn't seem there's much functionality change that would happen here).